hdmi/models/decoder_model.py

```diff
@@ -0,0 +1,250 @@
+from myhdl import Signal, intbv, always, always_comb, instances, block
+
+
+class DecoderModel:
+
+    def __init__(self, clock, data_in, video_preamble, data_island_preamble, c0,
+                 c1, vde, ade, video_out, audio_out, channel='BLUE'):
+
+        """
+         A non-convertible HDMI Decoder Model which decodes the TMDS data and outputs the video and aux data.
+         This is modelled after the xapp495 decoder module.
+
+        Args:
+            :param clock: The system clock or the pixel clock
+            :param data_in: The TMDS data (10 bits width) to be decoded
+            :param video_preamble: signal to detect the video preamble in the input data
+            :param data_island_preamble: signal to detect the data island preamble in the input data
+            :param c0: Control signal (hsync for Blue channel)
+            :param c1: Control signal (vsync for Blue channel)
+            :param vde: Video Data enable
+            :param ade: Audio data enable
+            :param video_out: Output video data
+            :param audio_out: Output audio (or aux) data.
+            :param channel: Color of the channel ('RED', 'GREEN' or 'BLUE'). Default value is 'BLUE'
```

Comment:
This might not have been discussed before but lets use the *Google* style comments.  It is pretty much what you have without the `:param` and ending `:`.  Optional you can put the expected type.

http://sphinxcontrib-napoleon.readthedocs.io/en/latest/index.html#google-vs-numpy